File: aa_lab_03/report/inc/lst/shaker.pas
procedure sort_shaker(var arr: array of Integer);
begin
  m := Length(arr)
  i := 0
  while i < m do
  begin
    j := i + 1
    while j < m do
    begin
      if arr[j] < arr[j-1] then
      begin
        Swap(arr[j], arr[j-1])
      end
      j := j + 1
    end
    m := m - 1
    k := m - 1
    while k > i do
    begin
      if arr[k] < arr[k-1] then
      begin
        Swap(arr[k], arr[k-1])
      end
      k := k - 1
    end
    i := i + 1
  end
end

File: aa_lab_03/report/inc/lst/shaker.cpp
void sort_shaker(vector<int> &arr)
{
  int m = arr.size();
   int i, j, k;
   for(i = 0; i < m;) {
      for(j = i+1; j < m; j++) {
         if(arr[j] < arr[j-1])
            swap(&arr[j], &arr[j-1]);
      }
      m--;
      for(k = m-1; k > i; k--) {
         if(arr[k] < arr[k-1])
            swap(&arr[k], &arr[k-1]);
      }
      i++;
   }
}